src/ssl/bio.cc

```diff
@@ -286,7 +308,7 @@ Ssl::ServerBio::readAndGive(char *buf, const int size, BIO *table)
         return giveBuffered(buf, size);
 
     if (record_) {
-        const int result = readAndBuffer(table);
+        const int result = readAndBuffer(table, SQUID_TCP_SO_RCVBUF);
```

Comment:
Limit the read size to "size" when KTLS is enabled.

Reply:
https://github.com/squid-cache/squid/pull/1629/commits/fa5d875cfb5444425bcdaecdedafa8a0e2e998e2